### ISSUE TITLE
[Feedback] Way to easily ignore rules/files for pmd

### ISSUE THREAD

**Issue by jaelee125** [forcedotcom/code-analyzer#1969]
### Feedback

I have a bunch of false positives for ProtectSensitiveData rule for XML files because of naming conventions on Custom Objects and Custom fields. It would be nice if there was an easy to ignore rules for certain xml files.

### Are you interested in contributing to this project?

Yes, I'd like to contribute.

**Comment by aruntyagiTutu** [comment]
Thanks @jaelee125 for sharing the issue.

`ProtectSensitiveData` is not a recommended rule by default. Also, we do support the `--target` parameter, which you can use to analyze only the files you’re interested in and exclude others as needed.

To understand it better can you please share more information, which client you are using and how ?

**Comment by namrata111f** [comment]
@jaelee125 We can also configure PMD XML file with the option to ignore the false positives:
```
<?xml version="1.0" encoding="UTF-8"?>
<ruleset name="Per-File Rule Suppression">

    <!-- ProtectSensitiveData: Suppress ONLY for specific files -->
    <rule ref="category/apex/security.xml/ProtectSensitiveData">
        <properties>
            <property name="violationSuppressXPath" 
                      value="//Document[contains(@FileName, 'SSN_Field__c') or 
                                        contains(@FileName, 'Password_Object__c') or
                                        contains(@FileName, 'Legacy_Sensitive')]" />
        </properties>
    </rule>
</ruleset>
``` 
You can explore more on this here: https://pmd.github.io/pmd/pmd_userdocs_suppressing_warnings.html 
Please re-open if this rules set suppression config is not working you.
